packages/relay/src/services/PostService.ts

```diff
@@ -1,29 +1,102 @@
 import { DB } from 'anondb'
-import { LOAD_POST_COUNT } from '../config'
+import {
+    DAY_DIFF_STAEMENT,
+    DB_PATH,
+    LOAD_POST_COUNT,
+    UPDATE_POST_ORDER_INTERVAL,
+} from '../config'
 import { UnirepSocialSynchronizer } from '../synchornizer'
 import { Helia } from 'helia'
 import { SnarkProof } from '@unirep/utils'
 import { epochKeyService } from './EpochKeyService'
 import { addActionCount } from '../utils/TransactionHelper'
 import { Post } from '../types/Post'
 import { ipfsService } from './IpfsService'
+import { PostgresConnector, SQLiteConnector } from 'anondb/node'
 
 export class PostService {
+    private cache: Post[] = []
+
+    async start(db: DB): Promise<void> {
+        const rows = await db.count('Post', {})
+        if (rows > 0) {
+            // fetch all posts during the initalization
+            await this.updateOrder(db)
+        }
+
+        // update the order of post table in the given interval
+        setInterval(async () => {
+            await this.updateOrder(db)
+        }, UPDATE_POST_ORDER_INTERVAL)
+    }
+
+    async updateOrder(db: DB): Promise<void> {
+        //      if user just posted, get the first ten result from db
+        //      pop last one and insert new post to the first element
+
+        //      1. The post that the user just posted
+        //      2. Follow the equation and sort them descendingly.
+        //         How many days before * (-0.5) + upvote count * 0.2 - downvote count * 0.2 + comment count * 0.1
+        //         a. good post 2 days ago（50 upvotes, 2 down votes, 14 comments）= 10
+        //         b. bad post just posted（10 upvotes, 17 downvotes, 8 comments）= -0.6
+        //      3. The posts >= 3 days ago should no longer follow this rule, but be sorted at the bottom based on the logic below
+        //         a. The post with the most upvotes and comments of the day
```

Comment:
I am not sure we cover the second part about "the day".
Maybe we should modify the way to record voteSum and commentCount, since we need to know vote and comment number everyday.
Should we reset the voteSum and commentCount everyday, once the post is larger than 3 days?

Below is SQL reference (Optimize by chatGPT XD, but better understand)
```
SELECT 
    *,
    ROW_NUMBER() OVER (
        PARTITION BY 
            CASE 
                WHEN ${DAY_DIFF_STATEMENT} <= 2 THEN 1
                ELSE 2
            END
        ORDER BY 
            CASE 
                WHEN ${DAY_DIFF_STATEMENT} <= 2 THEN (${DAY_DIFF_STATEMENT} * (-0.5)) + (upCount * 0.2) - (downCount * 0.2) + (commentCount * 0.1)
                ELSE (upCount + commentCount)
            END DESC,
            CASE 
                WHEN ${DAY_DIFF_STATEMENT} > 2 THEN upCount
                ELSE NULL
            END DESC,
            CASE 
                WHEN ${DAY_DIFF_STATEMENT} > 2 THEN commentCount
                ELSE NULL
            END DESC,
            publishedAt DESC
    ) AS OD
FROM 
    Post
ORDER BY 
    OD;
```